### PR TITLE
Move `build` workflow from CircleCI to GitHub Actions

### PR DESCRIPTION
**Pull request by rochdev**
### What does this PR do?

Move `build` workflow from CircleCI to GitHub Actions.

### Motivation

There are several advantages of switching to GitHub Actions for this type of project:

* Built-in action to easily and quickly switch Node version within a job. This means we can significantly reduce the parallelism of the jobs and use larger machines which are less likely to fail. It also means that we are back under the Codecov quota of 100 reports per commit, so we no longer need to merge reports manually.
* Built-in action to test ARM with Qemu. This is difficult in CircleCI since the type of runner for ARM is completely different.
* Reusable actions that can be pulled from other repositories. CircleCI orbs have to be published to a registry.
* Reusable workflows and JavaScript actions, both of which don't have an equivalent in CircleCI.
* Integrated directly in GitHub.
* Much larger ecosystem at this point.
* Windows and Mac machines start a lot faster.
* To my knowledge everyone on the team wants this change.

This PR focused only on the `build` workflow which is now split into separate workflows:

* **Project:** This is for any jobs for the entire projects, for example static analysis or integration tests.
* **Tracing:** This is for any jobs related only to tracing. Right now it was translated directly from CircleCI so it also includes profiling which we should move out later.
* **Plugins:** This is for our integrations. We could also move the jobs under `Tracing` but for now I felt like it was easier to use to group them in a single workflow.
* **Core:** This is for tests for common components that can be used by any product.

Upstream tests and benchmarks will be done in separate PRs.

### Additional Notes

* Cucumber is also failing on `master` as there was a new version released which breaks.
* I had to bump `pg` minimum version to make it pass, but this shouldn't be an issue since older versions only support up to Node 4 which we don't support .

### CODE REVIEW

File: docs/yarn.lock [hunk 1]
```diff
@@ -0,0 +1,244 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (bengl) [inline docs/yarn.lock:1]: Why are you removing this from gitignore and adding it here?

**Review by bengl — CHANGES_REQUESTED** [review]

File: docs/yarn.lock (hunk 1 above)
Comment (rochdev) [reply docs/yarn.lock:1]: This is needed by the `@actions/setup-node` action.

Comment (rochdev) [reply docs/yarn.lock:1]: Actually the action only needs to root lockfile, I can remove that one.

File: package.json [hunk 2]
```diff
@@ -110,8 +110,6 @@
     "jszip": "^3.5.0",
     "mkdirp": "^0.5.1",
     "mocha": "8",
-    "mocha-junit-reporter": "^2.0.0",
```
Comment (juan-fernandez) [inline package.json:113]: we don't want junit reports anymore? I think it's fine to remove them here. I can add them back later on if we think they can be useful

Comment (rochdev) [reply package.json:113]: CircleCI is able to use them in the UI, but GitHub Actions doesn't have such functionality. If this is something you're using locally I can keep bring it back.

Comment (juan-fernandez) [reply package.json:114]: feel free to leave it out. I can add it back later to dogfood ci visibility

File: docs/yarn.lock (hunk 1 above)
Comment (rochdev) [reply docs/yarn.lock:1]: As discussed offline, we can make sure this file is constantly updated with Dependabot.

**Review by bengl — APPROVED** [review]

**Pull request merged**